### PR TITLE
[NSX] Add SNAT support

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR adds support for SNAT on NSX VPCs

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

File: plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/service/NsxElement.java [hunk 1]
```diff
@@ -293,7 +296,10 @@ public boolean implementVpc(Vpc vpc, DeployDestination dest, ReservationContext
         }
         Account account = isNsxAndAccount.second();
         DomainVO domain = getDomainFromAccount(account);
-        return nsxService.createVpcNetwork(vpc.getZoneId(), account.getId(), domain.getId(), vpc.getId(), vpc.getName());
+        Network.Service[] services = { Network.Service.SourceNat };
+        boolean sourceNatEnabled = vpcOfferingServiceMapDao.areServicesSupportedByVpcOffering(vpc.getVpcOfferingId(), services);
+        // return nsxService.createVpcNetwork(vpc.getZoneId(), account.getId(), domain.getId(), vpc.getId(), vpc.getName(), sourceNatEnabled);
```
Comment (Pearl1594) [inline plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/service/NsxElement.java:301]: clean up commented code

**Comment by nvazquez** [comment]
@blueorangutan package

**Pull request merged**